test/integration/targets/ec2_group/tasks/main.yml

```diff
@@ -288,12 +288,15 @@
               cidr_ipv6: "64:ff9b::/96"
             <<: *aws_connection_info
           check_mode: true
+          diff: true
           register: result
 
         - name: assert state=present (expected changed=true)
           assert:
             that:
                - 'result.changed'
+               - 'result.diff.0.before.ip_permissions == result.diff.0.after.ip_permissions'
```

Comment:
Ideally this wouldn't be in the diff but we can remove that at a later date

Reply:
Okay, I'm happy improving it as we go along. It isn't in the visual diff, so I wasn't sure how you felt about that. Do you want to make our own system rather than use the callback's approach of showing only the changes with a little context interspersed between the "files"/resources?

Reply:
If it's not in the visual diff that should be fine - as you say, we can improve it later if we need to.

For all my nitpicking on this PR, this is an excellent improvement, thanks!